### PR TITLE
allow float fill for integer images in F.pad

### LINKED ISSUE #7948: Warning for fill stuff
```
pytest test/test_transforms_v2_refactored.py

================================================================ warnings summary =================================================================
test/test_transforms_v2_refactored.py: 34 warnings
  /home/nicolashug/dev/vision/torchvision/transforms/v2/functional/_geometry.py:1238: DeprecationWarning: an integer is required (got type float).  Implicit conversion to integers using __int__ is deprecated, and may be removed in a future version of Python.
    fill = torch.tensor(fill, dtype=image.dtype, device=image.device).reshape(-1, 1, 1)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html

```

### PR DESCRIPTION
**Pull request by pmeier**
Fixes #7948

cc @vfdev-5

### CODE REVIEW

File: test/test_transforms_v2_refactored.py [hunk 1]
```diff
@@ -417,6 +418,10 @@ def affine_bounding_boxes(bounding_boxes):
     )
 
 
+# turns all warnings into errors for this module
+pytestmark = pytest.mark.filterwarnings("error")
```
Comment (pmeier) [inline test/test_transforms_v2_refactored.py:422]: This makes sure, we never miss a warning coming from the transforms again.

File: test/test_transforms_v2_refactored.py [hunk 2]
```diff
@@ -312,11 +312,12 @@ def adapt_fill(value, *, dtype):
         return value
 
     max_value = get_max_value(dtype)
+    value_type = float if dtype.is_floating_point else int
```
Comment (pmeier) [inline test/test_transforms_v2_refactored.py:315]: If we adapt the fill value, we also want to adapt the type. I've checked and this does not hide the errors that we are fixing here.

File: test/test_transforms_v2_refactored.py [hunk 3]
```diff
@@ -2578,14 +2583,18 @@ def test_transform(self, param, value, make_input):
...
+        if param != "size":
             # 1. size is required
             # 2. the fill parameter only has an affect if we need padding
             kwargs["size"] = [s + 4 for s in self.INPUT_SIZE]
-
+        if param == "fill":
             if isinstance(input, tv_tensors.Mask) and isinstance(value, (tuple, list)):
                 pytest.skip("F.pad_mask doesn't support non-scalar fill.")
 
+            kwargs["fill"] = adapt_fill(
```
Comment (pmeier) [inline test/test_transforms_v2_refactored.py:2594]: I've missed to adapt the value here.

File: test/test_transforms_v2_refactored.py [hunk 4]
```diff
@@ -3478,6 +3487,8 @@ def test_transform_errors(self):
     def test_image_correctness(self, padding, padding_mode, fill, fn):
         image = make_image(dtype=torch.uint8, device="cpu")
 
+        fill = adapt_fill(fill, dtype=torch.uint8)
```
Comment (pmeier) [inline test/test_transforms_v2_refactored.py:3490]: Same as above

File: test/test_transforms_v2_refactored.py [hunk 5]
```diff
@@ -2578,14 +2583,18 @@ def test_transform(self, param, value, make_input):
         input = make_input(self.INPUT_SIZE)
 
         kwargs = {param: value}
-        if param == "fill":
+        if param != "size":
```
Comment (NicolasHug) [inline test/test_transforms_v2_refactored.py:2586]: Might need to relocate the `#1 ... # 2` comments below?

**Review by NicolasHug — APPROVED** [review]
Thanks Philip!

**Pull request merged**